### ISSUE TITLE
Compiler Gym results reproduction

### ISSUE THREAD

**Issue by MohamedAlmaki** [facebookresearch/CompilerGym#637]
## ❓ Questions and Help
I was unable to reproduce the same results in Table VI in [compiler gym paper](https://arxiv.org/abs/2109.08267) using the [reproduction code example](https://github.com/facebookresearch/CompilerGym/tree/development/examples/llvm_rl). The command used to run the experiment: 
```
python -m llvm_rl.train -m \
    experiment=cgym_repro \
    agent=ppo \
    num_replicas=1 \
    training=csmith \
    testing=all
```
I used the latest compiler gym version (0.2.3) on the latest development branch to run this experiment. 
When I tested the agent I got the following results: 
```
csmith: 0.849
cbench: 0.701
chstone: 0.871
```
But in the paper the results are: 
```
csmith: 1.245
cbench: 0.964 
chstone: 1.014
```
Note: this is calculated as `(instruction_count_init - instruction_count_final) / max((instruction_count_init - instruction_count_oz), 1)`. Unlike in the example code the instruction count reduction is calculated as `instruction_count_oz/max(instruction_count_final, 1)` in [line 110](https://github.com/facebookresearch/CompilerGym/blob/development/examples/llvm_rl/model/inference_result.py#L110). 

You can find the graphs for the experiment [here](https://tensorboard.dev/experiment/OfDZU2oJQVmBf9YpvrlM6Q/). I noticed that the evaluation reward mean is around 1.1. So it is clear that it will not achieve 1.245 when testing on csmith. 

Are there any additional things I need to do to reproduce the results? 

## Aditional Context 
Machine specs: 
RAM: 16 GB
CPU: Intel(R) Core(TM) i7-3960X CPU @ 3.30GHz (6 cores)

**Comment by ChrisCummins** [comment]
Hey @MohamedAlmaki, okay, will look into this! I took a quick scan through your tensorboard logs and don't notice anything unusual. 

> Note: this is calculated as (instruction_count_init - instruction_count_final) / max((instruction_count_init - instruction_count_oz), 1). Unlike in the example code the instruction count reduction is calculated as instruction_count_oz/max(instruction_count_final, 1) in [line 110](https://github.com/facebookresearch/CompilerGym/blob/development/examples/llvm_rl/model/inference_result.py#L110).

Not sure I follow. Are both of the sets of results you report computed using the same reward? Why not use the one from the paper?

Cheers,
Chris

**Comment by MohamedAlmaki** [comment]
Hey @ChrisCummins,  

Thanks for your feedback. 

> Not sure I follow. Are both of the sets of results you report computed using the same reward? Why not use the one from the paper?

Yeah, we used the one from the paper. But it was just unclear that the code reproduces the same results and uses another reward than the one in the paper. 

Regards, 
Mohammed Almakki

**Comment by ChrisCummins** [comment]
Okay. I haven't looked at the training scripts since I open sourced them. I could try running them locally and seeing if that produces the same results as you're seeing

Cheers,
Chris
